fixed-price-sale/program/src/processor/change_market.rs

```diff
@@ -58,10 +58,7 @@ impl<'info> ChangeMarket<'info> {
             market.price = new_price;
         }
 
-        // Check is required, because we can overwrite existing value
-        if let Some(new_pieces_in_one_wallet) = new_pieces_in_one_wallet {
-            market.pieces_in_one_wallet = Some(new_pieces_in_one_wallet);
-        }
+        market.pieces_in_one_wallet = new_pieces_in_one_wallet;
```

Comment:
I think here make sense to change type of `new_pieces_in_one_wallet` variable. I mean it would be better declare it as `Option<Enum{Value(u64), Unlim}>`. It will save consistency and decrease chance accidentally change this field.

Reply:
Should type changing affect whole contract or just single instruction 'ChangeMarket'?

Reply:
oh, yes, forgot about backward compatibility, thanks
so I think we'll new instruction with different arguments and mark current instruction as deprecated
such as adding new one argument will also break current front-end